### PR TITLE
Enable all tests in HttpsConnectionFilterTests to run on all platforms

### PR DESCRIPTION
**Pull request by mikeharder**
- Move `HttpClientSlim.cs` to `test\shared`
- Change `HttpClientSlim` to static class to simplify calling code
- Add `HttpClientSlim.PostAsync()`

@halter73, @cesarbs

### CODE REVIEW

**Comment by mikeharder** [comment]
ping

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/project.json [hunk 1]
```diff
@@ -31,6 +31,11 @@
   },
   "buildOptions": {
     "allowUnsafe": true,
+    "compile": {
+      "include": [
+        "../shared/**/*.cs"
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/project.json]: Surprised you didn't put this in a comment test project. Not that I'm complaining :smile:

**Comment by halter73** [comment]
Did you mean to switch `PathBaseTests`, `RequestTests`, `ResponseTests` and `ThreadCountTests` too?

**Comment by halter73** [comment]
Nvm. I see this is only for tests that need to not validate ssl certs. :shipit:

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/project.json (hunk 1 above)
Comment (mikeharder) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/project.json]: HttpClientSlim might be moved to the aspnet/Testing repo in the future.

**Pull request merged**